pkg/docker/config/config.go

```diff
@@ -136,10 +136,16 @@ func getAuthFilePaths(sys *types.SystemContext) []authPath {
 		// Logging the error as a warning instead and moving on to pulling the image
 		logrus.Warnf("%v: Trying to pull image in the event that it is a public image.", err)
 	}
-	paths = append(paths,
-		authPath{path: filepath.Join(homedir.Get(), dockerHomePath), legacyFormat: false},
-		authPath{path: filepath.Join(homedir.Get(), dockerLegacyHomePath), legacyFormat: true},
-	)
+	if dockerConfig := os.Getenv("DOCKER_CONFIG"); dockerConfig != "" {
+		paths = append(paths,
+			authPath{path: filepath.Join(dockerConfig, "config.json"), legacyFormat: false},
+		)
+	} else {
+		paths = append(paths,
+			authPath{path: filepath.Join(homedir.Get(), dockerHomePath), legacyFormat: false},
+			authPath{path: filepath.Join(homedir.Get(), dockerLegacyHomePath), legacyFormat: true},
+		)
+	}
```

Comment:
No, the legacy file always uses `$HOME` AFAICS, and is not ignored when `DOCKER_CONFIG` is set.

Reply:
I don't know why someone would go through the effort of setting `DOCKER_CONFIG` but still want the legacy behavior. In any case, done.

Reply:
Sure, but we never want to end up researching why the two behave differently when we can easily prevent it before merging the PR.